aten/src/ATen/mps/MPSAllocator.mm

```diff
@@ -46,68 +62,76 @@
 
 bool MPSHeapAllocatorImpl::alloc_buffer(AllocParams& p)
 {
-  if (m_set_fraction && m_total_allocated_memory + p.size() > max_available_size())
+  if (m_max_total_allowed_size != std::numeric_limits<size_t>::max() &&
+      current_allocated_size() + p.size() > m_max_total_allowed_size)
```

Comment:
Lets move this to a helper function. You have used this below as well.

Reply:
I'll keep a note and will do that whenever we fix and enable the high watermark level ratio.